textattack/transformations/word_swap_contract.py

```diff
@@ -0,0 +1,94 @@
+from textattack.transformations import Transformation
+
+
+class WordSwapContract(Transformation):
+    """Transforms an input by performing contraction on recognized
+    combinations."""
+
+    reverse_contraction_map = {
+        "is not": "isn't",
+        "are not": "aren't",
+        "cannot": "can't",
+        "could not": "couldn't",
+        "did not": "didn't",
+        "does not": "doesn't",
+        "do not": "don't",
+        "had not": "hadn't",
+        "has not": "hasn't",
+        "have not": "haven't",
+        "he is": "he's",
+        "how did": "how'd",
+        "how is": "how's",
+        "I would": "I'd",
+        "I will": "I'll",
+        "I am": "I'm",
+        "i would": "i'd",
+        "i will": "i'll",
+        "i am": "i'm",
+        "it would": "it'd",
+        "it will": "it'll",
+        "it is": "it's",
+        "might not": "mightn't",
+        "must not": "mustn't",
+        "need not": "needn't",
+        "ought not": "oughtn't",
+        "shall not": "shan't",
+        "she would": "she'd",
+        "she will": "she'll",
+        "she is": "she's",
+        "should not": "shouldn't",
+        "that would": "that'd",
+        "that is": "that's",
+        "there would": "there'd",
+        "there is": "there's",
+        "they would": "they'd",
+        "they will": "they'll",
+        "they are": "they're",
+        "was not": "wasn't",
+        "we would": "we'd",
+        "we will": "we'll",
+        "we are": "we're",
+        "were not": "weren't",
+        "what are": "what're",
+        "what is": "what's",
+        "when is": "when's",
+        "where did": "where'd",
+        "where is": "where's",
+        "who will": "who'll",
+        "who is": "who's",
+        "who have": "who've",
+        "why is": "why's",
+        "will not": "won't",
+        "would not": "wouldn't",
+        "you would": "you'd",
+        "you will": "you'll",
+        "you are": "you're",
+    }
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        """Return all possible transformed sentences, each with one
+        contraction."""
+        transformed_texts = []
+
+        words = current_text.words
+        indices_to_modify = list(indices_to_modify)
+
+        # search for every 2-words combination in reverse_contraction_map
+        for idx in indices_to_modify[:-1]:
+            word = words[idx]
+
+            next_idx = indices_to_modify[indices_to_modify.index(idx) + 1]
+            next_word = words[next_idx]
```

Comment:
**Easy suggestion**

Instead of this `next_idx` thing, you can just do something like:

```
for i, word_idx in indices_to_modify[:-1]:
  word = words[word_idx]
  next_word_idx = indices_to_modify[i+1]
  next_word = words[next_word_idx]
```

And I think there's a weird case you're forgetting where words in `indices_to_modify` form a contraction, even though they're not actually consecutive words in a sentence. Like if the sentence was "You are all" and indices_to_modify was {0, 2} then you might try and replace "You all" with "you'll" even though the words are non-consecutive.

Anyway, I think you can get around this with a:

```
if next_word_idx != word_idx + 1:
  continue 
```

**Potentially better suggestion**

Another way to do it would be calling the cluster function (the thing you called `cluster_idx` in the other classes) on `indices_to_modify`. Then iterate through the clusters and, if they're length two, check to see if they are replaceable contractions. Just an idea, either route is good with me.